python_modules/libraries/dagster-dbt/dagster_dbt_tests/core/test_asset_checks.py

```diff
@@ -434,6 +435,35 @@ def test_materialize_checks_no_asset(
     assert len(result.get_asset_observation_events()) == 0
 
 
+def test_get_manifest_and_translator_only_checks(
+    test_asset_checks_manifest: Dict[str, Any],
+    dbt_commands: List[List[str]],
+):
+    my_translator = DagsterDbtTranslator()
+
+    @dbt_assets(manifest=test_asset_checks_manifest, dagster_dbt_translator=my_translator)
+    def my_dbt_assets(context, dbt: DbtCliResource):
+        for dbt_command in dbt_commands:
+            cli_invocation = dbt.cli(dbt_command, context=context)
+            yield from cli_invocation.stream()
+
+    assert len(list(my_dbt_assets.check_specs)) > 0
+
+    my_dbt_assets = AssetsDefinition(
+        **{
+            **my_dbt_assets.get_attributes_dict(),
+            "specs": [],
+            "keys_by_input_name": {},
+            "keys_by_output_name": {},
+            "selected_asset_keys": set(),
+        }
+    )
```

Comment:
This test seems contrived and doesn't simulate the behavior using the steps that a user would take to create a dbt `AssetsDefinition` with only check specs.

My understanding is that:
1. They create an `AssetsDefinition` using `@dbt_assets`. It has both `specs` and `check_specs`.
2. They create a selection to only execute `check_specs`.
3. The framework creates the subsetted `AssetsDefinition` using the selection.
4. That `AssetsDefinition` should be valid.

Reply:
Those steps actually currently already happen in a dbt test called `test_materialize_checks_no_asset`. It works fine in the current codebase, but is broken by this PR that avoids keeping around unselected specs: https://github.com/dagster-io/dagster/pull/23145.

Before that PR is merged, the only way to organically create the situation that this PR is fixing would be with a dbt project that has tests but no models. This test simulates such a project in a contrived way.

I could also be convinced to actually create such a project. If you think that's the best route, do you have pointers on a project I should start with and where to put it?

Reply:
> Before that PR is merged, the only way to organically create the situation that this PR is fixing would be with a dbt project that has tests but no models. This test simulates such a project in a contrived way.

It doesn't seem like this could ever happen in `dagster-dbt` since modeling check specs in dbt is dependent on having the model/seed/snapshot modeled in the first place as a spec.

See: https://github.com/dagster-io/dagster/blob/dfe1b2afe1dfea408e1be333c2ceecdd0703e6db/python_modules/libraries/dagster-dbt/dagster_dbt/asset_utils.py#L754-L758

Reply:
I thought you could run tests on sources? https://stackoverflow.com/questions/78265533/dbt-testing-for-the-source-tables

Reply:
It's blocked by https://github.com/dagster-io/dagster/issues/18641.

Reply:
Yeah let's do that